orthopairs/src/main/java/org/reactome/release/orthopairs/Main.java

```diff
@@ -0,0 +1,155 @@
+package org.reactome.release.orthopairs;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class Main 
+{
+    /* Orthopairs has been overhauled completely so that it now gets all Homology and Gene-Protein information from PANTHER (pantherdb.org) instead of Ensembl.
+     *
+     * In the past, Orthopairs was a 3-step process with the goal of getting Protein Homologs between the source species and all of Reactome's species:
+     *  1) Download Gene-Protein information all species in Reactome and Protein-Gene information for the source species (Human in Reactome) from Ensembl's Biomart (An unstable step)
+     *  2) Get Gene Homologs between source species and all species in Reactome from Ensembl Compara (A very long step)
+     *  3) With the species Gene-Protein file from Biomart and the Gene Homolog file from Compara, map out a Protein homolog file
+     *
+     *  In this updated version, all of this information comes straight from 2 flat files provided by PANTHER. This makes things much quicker and stabler.
+     *  The only twist to this though, is that not all of the species Gene IDs come from Ensembl, but rather there is some that come from organism-specific databases.
+     *  The solution has been to use alternative ID mapping files from these organism-specific databases to convert any non-Ensembl IDs to their Ensembl equivalent.
+     *
+     *  At the end, we have the same two files for each species: The Gene-Protein mapping file specific to the species, and the Protein Homolog file.
+     *  All Protein IDs are UniProt (instead of the mix of Ensembl and UniProt) and all Gene IDs match the format used by Ensembl.
+     *
+     *  These files are much smaller than the older version, since PANTHER annotates which homologs are the 'Least Diverged', meaning we can filter out many that add noise to the dataset.
+     *
+     */
+
+    private static final Logger logger = LogManager.getLogger();
+
+    public static void main( String[] args ) throws IOException, ParseException {
+
+        String pathToConfig = "";
+        String sourceMappingSpecies = "";
+        // If using an alternative source species, specify the 4-letter code as the second argument
+        //TODO: Better solution for this. What if user wants default config but to change source species?
```

Comment:
You could use something like the library JCommander for named command line arguments and I think you can specify defaults.

https://stackoverflow.com/a/3337023

Reply:
Good point. Specifying the  path to config and an alternate source species should not be dependent on each other.

Reply:
Yea the named arguments seemed like the way to go, just didn't know how to do them, thanks!

Reply:
There's actually seemingly endless ways to do it in Java!  

https://stackoverflow.com/a/7829772

JCommander looks like a good option, though.

Reply:
https://picocli.info/ might be a good option, too.  It has autocompletion and other neat features.